New schema: PdgItem, PdgDecay, PdgParticleList

mjkramer opened the pull request:
A set of updates to support the new database schema, including decays. All tests pass in Python 3.11 and 2.7.

juergberinger commented:
Thanks Matt, generally this looks very good. As discussed at the meeting yesterday, let's wait for the latest SQLite file with the changes we discussed. Other than that, I was wondering if PdgBranchingFraction.products, being a property, shouldn't return a list rather than an iterator. Or make it a method rather than a property.

In addition, we need docstrings for all new classes and methods.

mjkramer commented:
Hi Juerg, I've added the missing docstrings, renamed "products" to "decay_products", and updated to the latest rpp2023 sqlite file. I agree that it would be better not to have properties that return iterators. Given that the other data in a PdgBranchingFraction are provided as properties, I've kept "decay_products" a property and changed it to return a list. Same for PdgItem.particles.
